bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+local goipath="${1}"
+local file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
+             "${gopath}/src/${goipath}" \
+             "${prefix}${gopath}/src/${goipath}"; do
+  [[ ${file}/ == ${proot}/* ]] && file=".${file#${proot}}"
+done
+file="$(realpath -sm ${file})"
+local workdir="$(realpath -sm .)"
+if [[ ${file}/ == ${workdir}/* ]] ; then
+  local dest="${prefix}${gopath}/src/${goipath}${file#${workdir}}"
+else
+  local dest="${prefix}${file}"
+fi
+local destdir="$(dirname ${dest})"
+if [[ ! -d "${destdir}" ]] ; then
+  installfile "${goipath}" "${destdir#${prefix}}"
+fi
+if [[ (! -e $file) || (-d "${file}" && ! -L "${file}") ]] ; then
+  install -m 0755 -vd "${dest}"
+  if [[ -d "${file}" && ! -L "${file}" ]] ; then
+    find "${file}" -iname '*.md' -type f -maxdepth 1 | while read docfile ; do
```

Comment:
IINM, this will not case the top most *.md files if the top most directory is not included in the `golist` output. If that is the case we should extend the `otherfiles` list with top most *.md files (if they exist).

Reply:
It will also take the top md files as long as you create .goipath from there (the directory level where .goipath exists is passed to the routine too)

For any md file that exists outside the .goipath subtree, you can just declare it manually as doc (it does not need installation in %{gopath} anyway, it lives somewhere else.